### PR TITLE
[NetAppFiles] Anf 9085 update swagger rest api to 2020 11 01

### PR DESCRIPTION
**Pull request by audunn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/examples/Accounts_CreateOrUpdate.json
Comment (j5lim) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/examples/Accounts_CreateOrUpdate.json:6]: It should be 2020-11-01. Please correct it in all files.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (j5lim) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:1770]: Please set `x-ms-long-running-operation` and `x-ms-long-running-operation-options` since it is async.

https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-long-running-operation
RPC: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#updating-using-patch

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (j5lim) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2784]: In a new api version, Azure resource should support a top level property `systemData`.

https://armwiki.azurewebsites.net/api_contracts/ResourceSystemData.html

You can reference the one in common-type.
https://github.com/Azure/azure-rest-api-specs/blob/a16488bbd0d17551c729e22ea6ac6f7c6588a0a3/specification/common-types/resource-management/v2/types.json#L445

**Review by j5lim — CHANGES_REQUESTED** [review]

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2784]: Is a update to the swagger enough to support this or does hit require change on the RP service side or this is handeled by ARM?

Comment (j5lim) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2784]: It requires changes in the RP. The above link has details.

It is okay to update the swagger without the RP changes, but the RP should implement the feature before releasing the SDKs to public.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:1770]: We have `x-ms-long-running-operation` but they do not have  `x-ms-long-running-operation-options`.   Should we apply `x-ms-long-running-operation-options`  to them all as well?

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2788]: @j5lim  FYI the current dev version is 2021.02.01 and this backend change can be added in that but in the pipeline and scheduled for release are API versions 2020.11 and 2020.12.  We need to add those swagger API versions for Completeness. 
As this backend update may take some time to get implemented and released so we ask if is this change is strictly mandatory and so urgent that it blocks this update or if can we finalize the implementation in the backend and add in this in the next upcoming 2021.02 that is scheduled soon.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:333]: Why you have 'delete' as part of the description? Also, when is your service returning either a 201 or 202?
According to our RPC, for async PUT calls the recommendation is to return 201 or 200, for async Delete calls return 202.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:372]: Update the description, a Delete call does not create or update your resource.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:402]: Any reason to have two similar definitions? You also have: netAppAccount and if not mistaken the only difference is systemData which is readOnly so you should be ok to reuse it.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:333]: While this is a good catch a small note that this is unchanged in this version. I am not sure where this came from has been in for a long time now probably form the beginning. The service does indeed return 201 here, I would be happy to fix this.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:402]: Same as earlier comment this has been in from the beginning. Happy to change if it improves the spec. Only concern is if that would be a breaking change for SDK's generated?

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2794]: @j5lim  can you comment on this.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:422]: Looks like actual result is `HTTP/1.1 202 Accepted` will update.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:230]: adding

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:469]: adding this

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:571]: I am happy to improve the spec if necessary. Just a small note this has been like this from before my time in previous API-Version, and I cannot unfortunately not speak for the  reasoning.  Will update

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:608]: In this case this is not an option some properties are not patchable and thus not part of the patch schema

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:891]: Added

File: package-lock.json
Comment (jorgecotillo) [inline package-lock.json:781]: I'll leave to the swagger team to accept this, you might need to roll back this updated file - but nothing ARM related

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2907]: mark this as secret please:

https://armwiki.azurewebsites.net/rp_onboarding/process/property_design_best_practices.html#secrets

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:3003]: This also seems to be a secret, please annotate accordingly.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [inline specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:3041]: If is a new property consider this as an enum.

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (jorgecotillo) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:399]: I cannot speak for the SDK generation process but I think it would as the object is changing, if it was like this in your previous version I guess you can continue with it but consider updating it in a future version.

**Comment by jorgecotillo** [comment]
@audunn I think the only missing part is for you to reply the question about adding x-ms-secret. Since you didn't commit & push your base api (as the initial push) and then on top of that applied your new changes I don't have a base api to compare your updates with. 

For future PRs, please commit and push your base api (as is) as the first push then start making changes as future commits, this expedites the review process and we can focus only on the changes.

**Comment by audunn** [comment]
> commit & push

Hi @jorgecotillo, well that is interesting, thing is initially I was told you just needed an initial commit for base comparison and that is what we have been doing for the past few updates. That is, we have been creating an initial commit then applying changes in subsequent commits and then we push. That you need that commit pushed as well actually explains a lot and we will make sure to push that initial commit from now to create a good base to compare from . 
Thank you for clearing up that misunderstanding for us.

**Comment by jorgecotillo** [comment]
ARM signed off

**Comment by audunn** [comment]
@yungezz  Can we move forward with this?

File: specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json
Comment (audunn) [reply specification/netapp/resource-manager/Microsoft.NetApp/stable/2020-11-01/netapp.json:2804]: @j5lim @yungezz  can you comment on this?

**Comment by audunn** [comment]
@yungezz Can we move forward with this, are there any more issues to address?

**Comment by yungezz** [comment]
hi @msyyc, could you pls check impact to Python SDK? thanks

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
hi @audunn, is the change deployed to Prod already? thanks

**Comment by audunn** [comment]
> hi @audunn, is the change deployed to Prod already? thanks

@yungezz what do you mean?  The service yes.

**Comment by audunn** [comment]
Please hold off on merging for now we are looking into possible small issue with python SDK. Should be resolved soon.

**Comment by audunn** [comment]
@yungezz can we move forward with this?

**Comment by yungezz** [comment]
> @yungezz can we move forward with this?

once @msyyc confirmed impact ot Python SDK, it's ok to merge. talking with him offline.

**Comment by msyyc** [comment]
For python, the PR is OK

**Comment by yungezz** [comment]
hi @j5lim you're in `Change Request` status, could you pls approve the PR if no more concern? thanks

**Review by j5lim — APPROVED** [review]

**Comment by audunn** [comment]
@yungezz can we move forward with this?

**Comment by audunn** [comment]
@yungezz can we move forward with this issue that was found just after this was merged https://github.com/Azure/azure-rest-api-specs/pull/12852?

**Comment by yungezz** [comment]
> @yungezz can we move forward with this issue that was found just after this was merged #12852? No SDK's have been created or released yet.

add @jianyexi to help on #12852

**Pull request merged**
